presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraMetadata.java

```diff
@@ -86,17 +85,18 @@ public CassandraMetadata(CassandraConnectorId connectorId,
     @Override
     public List<String> listSchemaNames(ConnectorSession session)
     {
-        return schemaProvider.getAllSchemas();
+        return cassandraSession.getCaseSensitiveSchemaNames().stream()
+                .map(name -> name.toLowerCase(ENGLISH))
+                .collect(toImmutableList());
     }
 
     @Override
     public CassandraTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName)
     {
         requireNonNull(tableName, "tableName is null");
         try {
-            CassandraTableHandle tableHandle = schemaProvider.getTableHandle(tableName);
-            schemaProvider.getTable(tableHandle);
-            return tableHandle;
+            cassandraSession.getTable(tableName);
```

Comment:
Either that, or just get the CassandraTable -- getTableHandle() performs the requisite checks that the table exists.